extensions/amp-auto-lightbox/0.1/measure.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+
+/** @type {!WeakMap<!Window, !IntersectionObserver>} */
+const intersectionObserversByWin = new WeakMap();
+
+/** @type {!WeakMap<!Element, !Deferred>} */
+const measuringElements = new WeakMap();
+
+/**
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {!Element} element
+ * @return {!Promise<!ClientRect>}
+ */
+export function measure(ampdoc, element) {
```

Comment:
Should we promote this to `src/dom.js`?

Reply:
Sorry, was going to comment on that and forgot. @samouri has already created `src/utils/intersection` and I can reuse that. The only nit is that the `src/utils/intersection` uses an `IntersectionObserver{root:document}`, which triggers the polyfill path in more cases than a simple `{root:null}` that I do here. So I wasn't sure if it's worth reusing the existing util or rolling this as a single-use utility for the extension. LMK what you think.